server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/git/GitRepository.java

```diff
@@ -1590,6 +1599,44 @@ private void readUnlock() {
         rwLock.readLock().unlock();
     }
 
+    @VisibleForTesting
+    boolean writeLock(boolean directExecution) {
+        // Should not commit changes if gc is running
+        boolean hasGcLock;
+        if (directExecution) {
+            try {
+                // Waits only 10 seconds the gc lock to prevent a command worker queue from filing up
+                // due to a long gc, if the command was directly submitted to this server.
+                hasGcLock = gcLock.tryLock() || gcLock.tryLock(10, TimeUnit.SECONDS);
+            } catch (InterruptedException e) {
+                hasGcLock = false;
+            }
+        } else {
+            // Should not fail replaying a log due to a long gc.
+            // Because if a log is failed to replay, this replica will enter read-only mode.
+            gcLock.lock();
```

Comment:
Shoudn't we acquire this lock when this repository is closed?

Reply:
Ah.. `gcLock` is needed.